### ISSUE TITLE
sendConnect() calls err.Error() when err is guaranteed to be nil, causing a panic

### ISSUE THREAD

**Issue by mcqueary** [nats-io/nats#119]
Long story short, `sendConnect()` sends a `CONNECT` and a `PING`, and then flushes the bufio writer. If `err` is non-nil at any point, the function returns the error. Then it reads a line from the server, expecting either `PONG` or `-ERR`. If it's neither of these things, then `err.Error()` is checked for a TLS error. 

The problem is that `err` is guaranteed to be nil here, causing a nil pointer panic. You can test for this by simply setting `opts.Verbose = true` (see #118) in your client, then trying to connect. This will cause `gnatsd` to send `+OK` in response to the `CONNECT`, which is unexpected. 

I'd say let's just remove the check for "tls: ", as this code is currently unreachable. There is also (obviously) no unit test for this condition or for ErrSecureConnFailed. In light of this, do we even need this exception/error?

Here's the affected snippet:
```go
	// We expect a PONG
	if line != pongProto {
		// But it could be something else, like -ERR
		if strings.HasPrefix(line, _ERR_OP_) {
			return errors.New("nats: " + strings.TrimPrefix(line, _ERR_OP_))
		} else if strings.HasPrefix(err.Error(), "tls: ") {
			// Or a TLS error:
			return ErrSecureConnFailed
		}

		return errors.New("nats: " + line)
	}
```

**Comment by kozlovic** [comment]
Would replacing err.Error by line fix the problem? I introduced the previous 'if' statements when making the process logic synchronous. We have to check for possible errors returned by the server.

-- 
Ivan Kozlovic | Principal Engineer
http://github.com/kozlovic
Apcera Inc.

**Comment by derekcollison** [comment]
Let's put together a quick test that shows the problem as specified above and then fix it.

**Comment by mcqueary** [comment]
+1. I believe we need to fix both things -- how we are checking for error, and also make sure we account for verbose being on #118.

**Comment by kozlovic** [comment]
I introduced this one too by moving changes from commit 4b1d6619fb820c2da7318369195d3928a7f1d07f there (when making connect synchronous). Looks like the intent of the original code was that GO language would return this kind of error in a write/read call. So I wonder if we need to have these kind of checks in sendConnect() to return an appropriate ErrSecureConnFailed error?

**Comment by kozlovic** [comment]
Ok, so this code needs to be removed for sure. The question is: should I "convert" any error returned in `sendConnect` function to `ErrSecureConnFailed` if the error starts with "tls: ", or do nothing and return the actual error returned by any of the `nc.bw` or `br` calls as it is now (which may be more valuable in understanding what is wrong)?

**Comment by mcqueary** [comment]
IMO, the latter. Return the actual error returned by `br`. More information is better. I don't believe we know (right now) every possible error that may be returned, so I wouldn't limit it to those starting with `"tls: "`.

**Comment by kozlovic** [comment]
I agree, I will submit a PR with the code (and new error that would not be used) removed. Will wait for Derek to get his opinion.

**Comment by kozlovic** [comment]
Fixed with #121
